### PR TITLE
tendermint: update to master upstream

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

update to latest master

Closes: #20

### CODE REVIEW

**Comment by musalbas** [comment]
How come tests are failing?

**Comment by marbar3778** [comment]
> How come tests are failing?

which tests?

proto-breakage will fail every time there is a breaking change made to a proto file, and the super linter covers markdown, yaml and openapi. I can look into that one.

**Comment by liamsi** [comment]
> How come tests are failing? 

Could be due to some small bug introduced when resolving merge conflicts. @marbar3778 will update to the latest master (the branch was based on the state of beginning of august and a lot happened since then). I'll review in a bit. 

Thanks so much for doing this Marko 💚

**Comment by liamsi** [comment]
The proto breakage fails due to the renaming:
`proto/tendermint/abci/types.proto:4:1:File option "go_package" changed from "github.com/tendermint/tendermint/abci/types" to "github.com/lazyledger/lazyledger-core/abci/types".`

I guess that is expected.

**Comment by liamsi** [comment]
Looks like the only lint that is still failing is about line length in markdown files (https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md013---line-length). I might simply disable it as the other LL repos do not impose a line length on md files anyways.

File: .markdownlint.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 default: true,
 MD007: { "indent": 4 }
-MD013: false
+MD013: true
```
Comment (liamsi) [inline .markdownlint.yml:3]: thanks

**Review by liamsi — APPROVED** [review]
Awesome, thanks a lot! 🎉 


yay

**Comment by liamsi** [comment]
The test that failed now `TestSignerVoteResetDeadline` isn't tracked in tendermint. Looks like a non-determinist error as we just played around with linter settings in the last few PRs...

**Comment by musalbas** [comment]
Thanks for this!

**Pull request merged**